internal/testutils/bubblewrap.go

```diff
@@ -0,0 +1,182 @@
+package testutils
+
+import (
+	"bytes"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	bubbleWrapSupportsUnprivilegedNamespacesOnce sync.Once
+	bubbleWrapSupportsUnprivilegedNamespaces     bool
+
+	bubbleWrapNeedsSudoOnce sync.Once
+	bubbleWrapNeedsSudo     bool
+)
+
+// CanRunBubbleWrapTest checks whether we can run tests running in bubblewrap or
+// sip the tests otherwise.
+func CanRunBubbleWrapTest(t *testing.T) {
+	t.Helper()
+
+	_, err := exec.LookPath("bwrap")
+	require.NoError(t, err, "Setup: bwrap cannot be found")
+
+	if os.Geteuid() == 0 {
+		t.Log("Running as EUID 0")
+		return
+	}
+
+	bubbleWrapSupportsUnprivilegedNamespacesOnce.Do(func() {
+		bubbleWrapSupportsUnprivilegedNamespaces = canUseUnprivilegedUserNamespaces(t)
+	})
+	if bubbleWrapSupportsUnprivilegedNamespaces {
+		t.Log("Can use unprivileged user namespaces")
+		return
+	}
+
+	bubbleWrapNeedsSudoOnce.Do(func() {
+		bubbleWrapNeedsSudo = canUseSudoNonInteractively(t)
+	})
+	if bubbleWrapNeedsSudo {
+		t.Log("Can use sudo non-interactively")
+	}
+
+	t.Skip("Skipping test: requires root privileges or unprivileged user namespaces")
+}
+
+// RunInBubbleWrapWithEnv runs the passed commands in bubble wrap sandbox with env variables.
+func RunInBubbleWrapWithEnv(t *testing.T, testDataPath string, env []string, args ...string) (string, error) {
+	t.Helper()
+
+	CanRunBubbleWrapTest(t)
+	return runInBubbleWrap(t, bubbleWrapNeedsSudo, testDataPath, env, args...)
+}
+
+// RunInBubbleWrap runs the passed commands in bubble wrap sandbox.
+func RunInBubbleWrap(t *testing.T, testDataPath string, args ...string) (string, error) {
+	t.Helper()
+
+	return RunInBubbleWrapWithEnv(t, testDataPath, nil, args...)
+}
+
+func runInBubbleWrap(t *testing.T, withSudo bool, testDataPath string, env []string, args ...string) (string, error) {
+	t.Helper()
+
+	var envArgs []string
+	if withSudo {
+		envArgs = append(envArgs, "sudo")
+	}
+	envArgs = append(envArgs,
+		"env",
+	)
+	if e := CoverDirEnv(); e != "" {
+		envArgs = append(envArgs, e)
+	}
+	envArgs = append(envArgs, env...)
+
+	etcDir := filepath.Join(testDataPath, "etc")
+	var etcPerms os.FileMode = 0700
+	if withSudo {
+		// If running in sudo we may want to give the folder more permissions
+		// not to lock out the caller.
+		etcPerms = 0777
```

Comment:
Since `/etc` has permissions 0755 and is also owned by root, there should be no need to make the `/etc` in the sandbox writable by non-root users, right?

```suggestion
		etcPerms = 0755
```

Reply:
Yeah, probably we can even avoid it now, but the problem we had on CI was with removal of the coverage artifacts.
However this happened when it was a subprocess, while I feel now it's not needed anymore so let me try to drop it completely.

Reply:
Ok, it works fine in CI too, so we can drop it now... This was needed when the directory was created by the wrapper script, but now it's handled all by the caller, so that's not needed.